### PR TITLE
Updated regex and form error message description in apply.jsx

### LINKED ISSUE #72: On application form, one-line intro field should allow `-` and `_`
image

### PR DESCRIPTION
**Pull request by ISPOL**
Changed regex to accept hyphens and underscores
Changed name regex to start with a letter.
Changed form control feedback to more closely match the regex requirements
Resolved #72

### CODE REVIEW

File: client/pages/apply.jsx [hunk 1]
```diff
@@ -78,11 +78,12 @@ class Apply extends React.Component {
                 type="text"
                 required
                 defaultValue={name}
-                pattern="[A-Za-z0-9 ]{3,50}"
+                pattern="[A-Za-z][A-Za-z0-9_\- ]{2,49}"
```
Comment (stain88) [inline client/pages/apply.jsx]: Do we want to keep a space here? Isn't that asking for trouble?

File: client/pages/apply.jsx [hunk 2]
```diff
@@ -78,11 +78,12 @@ class Apply extends React.Component {
...
                 data-error="Please enter"
               />
               <Form.Control.Feedback type="invalid">
-                Your name needs to be between 3 and 50 characters long.
+                Your name needs to be between 3 and 50 characters long and begin with a letter. Underscores and hyphens
```
Comment (stain88) [inline client/pages/apply.jsx]: I would change name to be between 2 and 50. A 2-letter name isn't that uncommon.

File: client/pages/apply.jsx [hunk 3]
```diff
@@ -93,10 +94,10 @@ class Apply extends React.Component {
                 type="text"
                 required
                 defaultValue={oneLineIntro}
-                pattern="[A-Za-z0-9,!?\x27\x22\s\.]{8,140}"
+                pattern="[A-Za-z0-9,!?_\-\x27\x22\s\.]{8,140}"
```
Comment (stain88) [inline client/pages/apply.jsx]: Anything against just using `\w`?

File: client/pages/apply.jsx (hunk 1 above)
Comment (ISPOL) [reply client/pages/apply.jsx]: I don't have a strong opinion on it, but people may want to type out their full names/pseudonyms and include spaces, e.g "Bruce Wayne" or "The Batman". The change makes it so "   " isn't valid anymore, at least.

File: client/pages/apply.jsx (hunk 2 above)
Comment (ISPOL) [reply client/pages/apply.jsx]: Can't disagree with that, will change.

File: client/pages/apply.jsx (hunk 3 above)
Comment (ISPOL) [reply client/pages/apply.jsx]: Just laziness. Iirc \w doesn't apply to hyphens, so it's a difference of adding three characters "_\\-" vs deleting a bunch of stuff and then adding four chars "\w\\-".

Comment (ISPOL) [reply client/pages/apply.jsx]: Next commit has it changed.

**Comment by lpatmo** [comment]
Thanks @stain88 and @ISPOL! Merging.

**Pull request merged**